File: frontend/src/types/agent.ts
export type AgentRunStatus = "queued" | "running" | "completed" | "failed" | "cancelled"

export type AgentRun = {
  id: string
  taskId: string
  agentType: string
  status: AgentRunStatus
  startedAt: string | null
  completedAt: string | null
  outputLog: string
  validationResult: {
    passed: boolean
    checks: { name: string; passed: boolean; message: string }[]
  } | null
  retryCount: number
}

export type DecisionLog = {
  id: string
  projectId: string
  decision: string
  reasoning: string
  alternatives: string[]
  madeBy: string
  createdAt: string
}

export type ChatMessage = {
  id: string
  role: "user" | "assistant" | "system"
  content: string
  timestamp: string
  metadata?: Record<string, unknown>
}

export type ChatSession = {
  id: string
  projectId: string
  userId: string
  messages: ChatMessage[]
  sessionType: "planning" | "refinement" | "architecture"
  createdAt: string
}

File: backend/app/schemas/agent.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class AgentDispatch(BaseModel):
    agent_type: str = Field(..., min_length=1, max_length=100)
    config: dict | None = None


class AgentRunResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    task_id: str | None
    agent_type: str
    status: str
    started_at: datetime | None
    completed_at: datetime | None
    output_log: str | None
    validation_result: dict | None
    retry_count: int
    created_at: datetime
    updated_at: datetime
